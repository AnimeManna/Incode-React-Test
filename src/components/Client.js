import React, {Component} from 'react'

import {connect} from 'react-redux'

import ListItem from './ListItem'

import {
    Item,
    Image,
    List,
    Segment
} from 'semantic-ui-react'

class Client extends Component{
    render(){
        console.log('USER ID',this.props.user.id)
        const {general} = this.props.user
        const {job} = this.props.user
        const {contact} = this.props.user
        const {address} = this.props.user
        return(
                <Segment>
                    <Item.Group style={{display:'flex', justifyContent:'flex-start'}}>
                        <Item.Image src={general.avatar} width="128" height="128" />
                        <Item.Content style={{padding:10}}>
                            <List>
                                <ListItem icon="user" firstText={general.firstName} secondText={general.lastName}  />
                                <ListItem icon="briefcase" firstText={job.company} secondText={job.title}  />
                                <ListItem icon="marker" firstText={address.country} secondText={address.city} thirdText={address.street} fourthText={address.zipCode} />
                                <ListItem icon="mail" firstText={contact.email} secondText={contact.phone} />
                            </List>
                        </Item.Content>
                    </Item.Group>
                </Segment>
        )
    }
}

export default connect(
    state => ({
        user: state.clientsReducer.user
    })
)(Client)