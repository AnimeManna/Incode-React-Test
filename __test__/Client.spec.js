import React from 'react'
import configureStore from 'redux-mock-store'
import renderer from 'react-test-renderer'
import {shallow, render, mount} from 'enzyme'

import Client from '../src/components/Client'
import toJson from "enzyme-to-json";

describe('Client components',()=>{
    const mockStore = configureStore()
    const initialState = {
        clientsReducer:{
            clients:[

            ],
            viewClients:[

            ],
            user: {
                general: {
                    firstName: 'Ololoev',
                    lastName: 'Ololevn',
                    avatar: 'https://s3.amazonaws.com/uifaces/faces/twitter/andysolomon/128.jpg'
                },
                job: {
                    title: 'DNU',
                    company: 'Incode-Group'
                },
                contact: {
                    email: 'ololoev@gmail.com',
                    phone: '0999201'
                },
                address: {
                    country: 'Ukraine',
                    city: 'Dnepr',
                    street: 'Kazakova',
                    zipCode: '49000'

                }
            }
        }
    }
    const store = mockStore(initialState);
    describe('render()', ()=>{
        test('View client info',()=>{
            const wrapper = shallow (<Client store={store} />)
            const component = wrapper.dive();
            expect(toJson(component)).toMatchSnapshot();
        })
    })
})