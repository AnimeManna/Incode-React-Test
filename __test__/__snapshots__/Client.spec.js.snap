// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Client components render() View client info 1`] = `
<Segment>
  <ItemGroup
    style={
      Object {
        "display": "flex",
        "justifyContent": "flex-start",
      }
    }
  >
    <ItemImage
      height="128"
      src="https://s3.amazonaws.com/uifaces/faces/twitter/andysolomon/128.jpg"
      width="128"
    />
    <ItemContent
      style={
        Object {
          "padding": 10,
        }
      }
    >
      <List>
        <ListItem
          firstText="Ololoev"
          icon="user"
          secondText="Ololevn"
        />
        <ListItem
          firstText="DNU"
          icon="briefcase"
          secondText="Incode-Group"
        />
        <ListItem
          firstText="Ukraine"
          fourthText="49000"
          icon="marker"
          secondText="Dnepr"
          thirdText="Kazakova"
        />
        <ListItem
          firstText="ololoev@gmail.com"
          icon="mail"
          secondText="0999201"
        />
      </List>
    </ItemContent>
  </ItemGroup>
</Segment>
`;
